utils/mgmt_system/google.py

```diff
@@ -0,0 +1,290 @@
+
+        if not source_disk_image:
+            source_disk_image = "projects/debian-cloud/global/images/debian-7-wheezy-v20150320"
+
+        machine_type = machine_type or ("zones/%s/machineTypes/n1-standard-1" % self._zone)
+
+        try:
+            script = open(startup_script, 'r').read()
+        except:
+            logger.debug("Couldn't open the startup script %s for GoogleCloudSystem"
+                        % startup_script)
+            script = "#!/bin/bash"
+
+        config = {
+            'name': instance_name,
+            'machineType': machine_type,
+
+            # Specify the boot disk and the image to use as a source.
+            'disks': [
+                {
+                    'boot': True,
+                    'autoDelete': True,
+                    'initializeParams': {
+                        'sourceImage': source_disk_image,
+                    }
+                }
+            ],
+
+            # Specify a network interface with NAT to access the public
+            # internet.
+            'networkInterfaces': [{
+                'network': 'global/networks/default',
+                'accessConfigs': [
+                    {'type': 'ONE_TO_ONE_NAT', 'name': 'External NAT'}
+                ]
+            }],
+
+            # Allow the instance to access cloud storage and logging.
+            'serviceAccounts': [{
+                'email': 'default',
+                'scopes': [
+                    'https://www.googleapis.com/auth/devstorage.read_write',
+                    'https://www.googleapis.com/auth/logging.write'
+                ]
+            }],
+
+            # Metadata is readable from the instance and allows you to
+            # pass configuration from deployment scripts to instances.
+            'metadata': {
+                'items': [{
+                    # Startup script is automatically executed by the
+                    # instance upon startup.
+                    'key': 'startup-script',
+                    'value': script
+                }, {
+                    # Every project has a default Cloud Storage bucket that's
+                    # the same name as the project.
+                    'key': 'bucket',
+                    'value': self._project
+                }]
+            }
+        }
+
+        operation = self._compute.instances().insert(
+            project=self._project, zone=self._zone, body=config).execute()
+        wait_for(lambda: self._nested_wait_vm_running(operation['name']), delay=0.5,
+            num_sec=timeout, message=" Create %s" % instance_name)
+        return True
+
+    def delete_vm(self, instance_name, timeout=180):
+        if not self.does_vm_exist(instance_name):
+            logger.info(" The %s instance is not exists, skipping" % instance_name)
+            return True
+
+        logger.info(" Deleting Google Cloud instance %s" % instance_name)
+        operation = self._compute.instances().delete(
```

Comment:
Yes, we can re-use it. Done. 
Also I left the self._compute, now it is used for waiting procedure and also will be potentially used for templates in future.